indico/modules/events/editing/controllers/backend/editable_list.py

```diff
@@ -152,21 +151,23 @@ def _process(self, text, internal):
 
 class RHApplyJudgment(RHEditablesBase):
     @use_kwargs({
-        'action': fields.String(required=True, validate=validate.OneOf(['accept', 'reject']))
+        'action': EnumField(EditingReviewAction, required=True)
     })
     def _process(self, action):
+        if action == EditingReviewAction.update:
+            raise ValueError('Invalid action')
         changed = []
+        disallowed_state = {
+            EditingReviewAction.accept: EditableState.accepted,
+            EditingReviewAction.reject: EditableState.rejected,
+            EditingReviewAction.request_update: EditableState.needs_submitter_changes
+        }[action]
         for editable in self.editables:
-            revision = editable.latest_revision
-            if action == 'accept' and not revision.has_publishable_files:
+            if action == EditingReviewAction.accept and not editable.has_publishable_files:
                 continue
-            if revision.type == RevisionType.ready_for_review:
-                review_and_publish_editable(revision, EditingReviewAction[action], '')
-            elif revision.type == RevisionType.needs_submitter_confirmation and action == 'accept':
-                confirm_and_publish_changes(revision, EditingConfirmationAction.accept, '')
-                db.session.flush()
-            else:
+            if editable.state == disallowed_state:
                 continue
+            review_and_publish_editable(editable.latest_revision, action, management=True)
```

Comment:
Are we failing gracefully if you try to accept an editable that does not contain any publishable files?

Reply:
currently it just skips the paper and shows the general warning

![image](https://github.com/user-attachments/assets/5ea4c43a-e292-43d5-9dd2-8c1e2b12df2f)